### PR TITLE
Bump fuels to 0.50.1

### PR DESCRIPTION
**Pull request by DefiCake**

### CODE REVIEW

File: packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs [hunk 1]
```diff
@@ -338,6 +372,8 @@ mod success {
             .find(|&r| matches!(r, Receipt::MessageOut { .. }))
             .unwrap();
 
+        dbg!(hex::encode(message_receipt.data().unwrap()));
+
```
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs:376]:
```suggestion
```

File: packages/fungible-token/bridge-fungible-token/tests/utils/builder.rs [hunk 2]
```diff
@@ -41,35 +48,20 @@ pub async fn build_contract_message_tx(
 
     // Build a change output for the owner of the first provided coin input
     if !gas_coins.is_empty() {
```
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/utils/builder.rs:50]: No more outputs being added below?

File: packages/message-predicates/contract-message-predicate/tests/utils/builder.rs [hunk 3]
```diff
@@ -53,23 +53,79 @@ pub async fn build_contract_message_tx(
...
+        .with_tx_params(params)
+        .with_script(script_bytecode);
+
+    wallet.sign_transaction(&mut builder);
 
+    builder.build().unwrap()
+}
+
+/// Build a message-to-contract transaction with the given input coins and outputs, but invalid script bytecode
+/// note: unspent gas is returned to the owner of the first given gas input
```
Comment (Braqzen) [inline packages/message-predicates/contract-message-predicate/tests/utils/builder.rs:75]: I guess this is why the outputs were removed from above

File: packages/message-predicates/contract-message-predicate/tests/utils/builder.rs [hunk 4]
```diff
@@ -53,23 +53,79 @@ pub async fn build_contract_message_tx(
...
+/// Build a message-to-contract transaction with the given input coins and outputs
```
Comment (Braqzen) [inline packages/message-predicates/contract-message-predicate/tests/utils/builder.rs:76]: Which message should remain here?

**Review by Braqzen — COMMENTED** [review]
Is the hex crate still needed?

File: packages/fungible-token/bridge-fungible-token/tests/utils/builder.rs (hunk 2 above)
Comment (DefiCake) [reply packages/fungible-token/bridge-fungible-token/tests/utils/builder.rs:50]: No asset change generated in these file, so yeah, no outputs

File: packages/message-predicates/contract-message-predicate/tests/utils/builder.rs (hunk 4 above)
Comment (DefiCake) [reply packages/message-predicates/contract-message-predicate/tests/utils/builder.rs]: Solved with the correct one

**Comment by DefiCake** [comment]
> Is the hex crate still needed?

I removed in from the fungible-token-tests, it is still used in message-predicates. I moved it to the root of the project and marked it to be used within the workspace, because I use it here and there when I need to debug

**Review by Braqzen — APPROVED** [review]

**Pull request merged**